lib/httparty/connection_adapter.rb

```diff
@@ -69,10 +70,11 @@ def initialize(uri, options = {})
 
     def connection
       host = clean_host(uri.host)
+      default_port = uri.scheme == 'https' ? 443 : 80
```

Comment:
Well, I tried to decouple from URI in the first commit and this was a biased change.
Addressable:URI sets port to nil when the url is pointing to a default port. Ruby's URI [sets it to 80/443](https://github.com/ruby/ruby/blob/trunk/lib/uri/generic.rb#L210) so that's why I've set the same fallback:

```ruby
require "minitest/autorun"
require "uri"
require "addressable/uri"

class TestRubyImplementations < Minitest::Unit::TestCase

  def test_uri_port_defaults
    url = 'http://foo/bar'
    uri = URI.parse url
    assert_equal uri.port, 80

    uri = Addressable::URI.parse url
    assert_equal uri.port, nil
  end

end
```